db/crud.go

```diff
@@ -725,6 +742,10 @@ func (db *Database) Put(docid string, body Body) (newRevID string, err error) {
 // and the new revision being added is "3-cde", then docHistory passed in should be ["2-bcd", "1-abc"].
 //
 func (db *Database) PutExistingRev(docid string, body Body, docHistory []string, noConflicts bool) error {
+	return db.PutExistingRevRoundTrip(docid, body, docHistory, noConflicts, false)
```

Comment:
As discussed, investigate whether it's possible to switch Put/PutExistingRev to return doc, so that the main code patch can remain where it is, and the RoundTrip methods can wrap that.